### PR TITLE
Nv20230821

### PR DESCRIPTION
**Pull request by nvolk**
- Move prepublicationUtils.js here (was in marc-record-merger-reducers-js before)
- Implement validator/fixer for MRA-273 (X00$a space between initials)
- Support field 946 (MET-461)
- improved subfield order between subfields $2 and $0 (solves eg. part of by MRA-465)
- Fix MRA-433 (If two therwise identical 490 fields exist, keep the one with ind1=1 and drop the one with ind1=0. Both lone fields and chained fields are supported)
- Fix MET-467 (chain: if better 7XX fields exists, remove worse 1XX field, and re-tag 7XX as 1XX)

### CODE REVIEW

**Comment by nvolk** [comment]
Forgot to mention usage of newer dependencies:

```
        "dependencies": {
                "@natlibfi/marc-record": "^7.3.0",
                "@natlibfi/marc-record-merge": "^7.0.0",
                "@natlibfi/marc-record-validate": "^8.0.1",
                "@natlibfi/marc-record-validators-melinda" : "git+ssh://git@github.com/natlibfi/marc-record-validators-melinda.git#nv20230821",
                "@natlibfi/melinda-commons": "^13.0.5",
                "clone": "^2.1.2",
                "debug": "^4.3.4",
                "isbn3": "^1.1.40",
                "normalize-diacritics": "^4.0.0"
        },
```

**Comment by nvolk** [comment]
Forgot to mention updated dependencies:
```
  "dependencies": {
    "@babel/register": "^7.22.5",
    "@natlibfi/issn-verify": "^1.0.3",
    "@natlibfi/marc-record": "^7.3.0",
    "@natlibfi/marc-record-validate": "^8.0.1",
    "cld3-asm": "^3.1.1",
    "debug": "^4.3.4",
    "isbn3": "^1.1.40",
    "langs": "^2.0.0",
    "node-fetch": "^2.6.12",
    "xml2js": ">=0.6.2 <1.0.0",
    "clone": "^2.1.2"
  },
```

File: src/punctuation2.js [hunk 1]
```diff
@@ -187,6 +187,17 @@ const cleanLegalSeriesTitle = [ // 490 and 830
   {'code': 'axyz', 'followedBy': 'v', 'remove': / *;$/u}
 ];
 
+const clean24X = [
+  {'name': 'I:A', 'code': 'i', 'followedBy': 'i', 'remove': / *:$/u},
```
Comment (ammsalme) [inline src/punctuation2.js:191]:
```suggestion
  {'name': 'I:A', 'code': 'i', 'followedBy': 'a', 'remove': / *:$/u},
```

Comment (ammsalme) [reply src/punctuation2.js:191]: Should be 'a' ?

File: test-fixtures/remove-inferior-datafields/f07a/expectedResult.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07a/expectedResult.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: test-fixtures/remove-inferior-datafields/f07a/record.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07a/record.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: test-fixtures/remove-inferior-datafields/f07b/expectedResult.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07b/expectedResult.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: test-fixtures/remove-inferior-datafields/f07b/record.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07b/record.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: test-fixtures/remove-inferior-datafields/f07c/expectedResult.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07c/expectedResult.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: test-fixtures/remove-inferior-datafields/f07c/record.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{
+  "_validationOptions": {},
+  "fields": [
+    { "tag": "001", "value": "f01" },
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
+      { "code": "a", "value": "foo :"},
+      { "code": "b", "value": "bar /"},
+      { "code": "c", "value": "Kerho Ukkonen."}
+    ]},
+    { "tag": "245", "ind1": "1", "ind2": " ", "subfields": [
```
Comment (ammsalme) [inline test-fixtures/remove-inferior-datafields/f07c/record.json:10]:
```suggestion
    { "tag": "246", "ind1": "1", "ind2": " ", "subfields": [
```

File: src/punctuation2.js (hunk 1 above)
Comment (nvolk) [reply src/punctuation2.js:191]: It should indeed...

**Review by ammsalme — APPROVED** [review]

**Pull request merged**